shared/ghutil/fakeghutil/fakeghutil.go

```diff
@@ -185,6 +189,134 @@ func (fgc *FakeGithubClient) RemoveLabelForIssue(org, repo string, issueNumber i
 	return nil
 }
 
+// ListPullRequests lists pull requests within given repo, filters by head user and branch name if
+// provided as "user:ref-name", and by base name if provided, i.e. "master"
+func (fgc *FakeGithubClient) ListPullRequests(org, repo, head, base string) ([]*github.PullRequest, error) {
+	var res []*github.PullRequest
+	PRs, ok := fgc.PullRequests[repo]
+	if !ok {
+		return res, fmt.Errorf("repo %s not exist", repo)
+	}
+	for _, PR := range PRs {
+		// Filter with consistent logic of CreatePullRequest function below
+		if ("" == head || *PR.Head.Label == head) &&
+			("" == base || *PR.Base.Ref == base) {
+			res = append(res, PR)
+		}
+	}
+	return res, nil
+}
+
+// ListCommits lists commits from a pull request
+func (fgc *FakeGithubClient) ListCommits(org, repo string, ID int) ([]*github.RepositoryCommit, error) {
+	commits, ok := fgc.PRCommits[ID]
+	if !ok {
+		return commits, fmt.Errorf("no commits found for PR '%d'", ID)
+	}
+	return commits, nil
+}
+
+// ListFiles lists files from a pull request
+func (fgc *FakeGithubClient) ListFiles(org, repo string, ID int) ([]*github.CommitFile, error) {
+	var res []*github.CommitFile
+	commits, err := fgc.ListCommits(org, repo, ID)
+	if nil != err {
+		return res, err
+	}
+	for _, commit := range commits {
+		files, ok := fgc.CommitFiles[*commit.SHA]
+		if !ok {
+			// don't allow empty commit
+			return res, fmt.Errorf("no files found for commit '%s'", *commit.SHA)
+		}
+		res = append(res, files...)
+	}
+	return res, nil
+}
+
+// CreatePullRequest creates PullRequest, passing head user and branch name "user:ref-name", and base branch name like "master"
+func (fgc *FakeGithubClient) CreatePullRequest(org, repo, head, base, title, body string) (*github.PullRequest, error) {
+	PRNumber := fgc.getNextNumber()
+	stateStr := string(ghutil.PullRequestOpenState)
+	b := true
+	PR := &github.PullRequest{
+		Title:               &title,
+		Body:                &body,
+		MaintainerCanModify: &b,
+		State:               &stateStr,
+	}
+	if "" != head {
+		tokens := strings.Split(head, ":")
+		if len(tokens) != 2 {
+			return nil, fmt.Errorf("invalid head, want: 'user:ref', got: '%s'", head)
+		}
+		PR.Head = &github.PullRequestBranch{
+			Label: &head,
+			Ref:   &tokens[1],
+		}
+	}
+	if "" != base {
+		l := fmt.Sprintf("%s:%s", repo, base)
+		PR.Base = &github.PullRequestBranch{
+			Label: &l,
+			Ref:   &base,
+		}
+	}
+
+	if _, ok := fgc.PullRequests[repo]; !ok {
+		fgc.PullRequests[repo] = make(map[int]*github.PullRequest)
+	}
+	fgc.PullRequests[repo][PRNumber] = PR
```

Comment:
should this be wiuthin if? If the if conditon fails, you will not be creating the map and go will panic

Reply:
it would not be added if the repo already exist if moved within